compiler/modules/nback/build.bal

```diff
@@ -948,8 +948,210 @@ function buildCompare(llvm:Builder builder, Scaffold scaffold, bir:CompareInsn i
+                        nullableValue = lhsVal;
+                        other = rhs;
+                        otherValue = rhsVal;
+                    }
+                    else {
+                        nullableOperand = rhs;
+                        nullableValue = rhsVal;
+                        other = lhs;
+                        otherValue = lhsVal;
+                    }
+                    llvm:Value isNull = builder.iCmp("eq", nullableValue, llvm:constNull(llvm:pointerType("i8", 1)));
+                    llvm:BasicBlock isNullBB = scaffold.addBasicBlock();
+                    llvm:BasicBlock notNullBB = scaffold.addBasicBlock();
+                    llvm:BasicBlock joinBB = scaffold.addBasicBlock();
+                    builder.condBr(isNull, isNullBB, notNullBB);
+
+                    builder.positionAtEnd(isNullBB);
+                    buildStoreBoolean(builder, scaffold, llvm:constInt(LLVM_BOOLEAN,0), insn.result);
+                    builder.br(joinBB);
+
+                    builder.positionAtEnd(notNullBB);
+                    if lhs is () || rhs is () {
+                        builder.unreachable();
+                    }
+                    else {
+                        check untagAndCompare(builder, scaffold, insn);
+                        builder.br(joinBB);
+                    }
+                    
+                    builder.positionAtEnd(joinBB);
+                }
+            }
+            else {
+                return err:unimplemented("compare with operands of non-optional unimplemented type");
+            }
+        }
+    }
+}
+
+function loadOperandForOptionalComparison(llvm:Builder builder, Scaffold scaffold, bir:CompareInsn insn, bir:Operand operand) returns llvm:Value|BuildError {
+    bir:OrderType ot = insn.orderType;
+    if ot is bir:OptOrderType {
+        if operand is () {
+            return llvm:constNull(llvm:pointerType("i8", 1));
+        }
+        else {
+            llvm:Value? val = ();
+            match ot.opt {
+                t:UT_INT => {
+                    val = buildInt(builder, scaffold, <bir:IntOperand> operand);
+                }
+                t:UT_FLOAT => {
+                    val = buildFloat(builder, scaffold, <bir:FloatOperand> operand);
+                }
+                t:UT_BOOLEAN => {
+                    val = buildBoolean(builder, scaffold, <bir:BooleanOperand> operand);
+                }
+                t:UT_STRING => {
+                    val = check buildString(builder, scaffold, <bir:StringOperand> operand);
+                }
+            }
+            if val is () {
+                panic error("Unknown comparison type");
+            }
+            else {
+                return val;
+            }
+        }
+    }
+    else {
+        panic error("Not an optional order type");
+    }
+}
+
+function untagAndCompare(llvm:Builder builder, Scaffold scaffold, bir:CompareInsn insn)returns BuildError? {
```

Comment:
Use names consistent with the rest of file.

Reply:
Function removed